TicketBooking/Flight/FlightRepository/FlightRepository.cs

```diff
@@ -0,0 +1,174 @@
+using TicketBooking.Flight.Services;
+using TicketBooking.AppSettings;
+
+namespace TicketBooking.Flight.FlightRepository;
+
+public class FlightRepository
+{
+    private List<FlightModel.Flight> Flights { get; set; } = new List<FlightModel.Flight>();
+
+    public FlightRepository()
+    {
+        LoadFlights();
+
+        // Sure that flights data stored in the app file are all valid, so upload feedback cleared from the console
+        Console.Clear();
+    }
+
+    public List<Flight.FlightModel.Flight> FilterFlights(
+        double? priceFrom = null,
+        double? priceTo = null,
+        string? departureCountry = null,
+        string? destinationCountry = null,
+        string? departureAirport = null,
+        string? arrivalAirport = null,
+        ClassEnum? flightClass = null
+        )
+    {
+        List<Flight.FlightModel.Flight> tempFlights = Flights;
+
+        if (departureCountry is not null)
+        {
+            tempFlights = tempFlights.Where(flight => flight.DepartureCountry.Equals((CountryEnum)Enum.Parse(typeof(CountryEnum), departureCountry))).ToList();
+        }
+
+        if (destinationCountry is not null)
+        {
+            tempFlights = tempFlights.Where(flight => flight.DestinationCountry.Equals((CountryEnum)Enum.Parse(typeof(CountryEnum), destinationCountry))).ToList();
+        }
+
+        if (departureAirport is not null)
+        {
+            tempFlights = tempFlights.Where(flight => flight.DepartureAirport.Equals((AirportEnum)Enum.Parse(typeof(AirportEnum), departureAirport))).ToList();
+        }
+
+        if (arrivalAirport is not null)
+        {
+            tempFlights = tempFlights.Where(flight => flight.ArrivalAirport.Equals((AirportEnum)Enum.Parse(typeof(AirportEnum), arrivalAirport))).ToList();
+        }
+
+        if (flightClass is not null)
+        {
+            tempFlights = tempFlights
+                .Where(flight => flight.Class.Any(availableClass => availableClass.Key.Equals(flightClass)))
+                .ToList();
+        }
+
+        if (priceFrom is not null)
+        {
+            tempFlights = tempFlights.Where(flight => flight.Class.Any(availableClass => availableClass.Value >= priceFrom)).ToList();
+        }
+
+        if (priceTo is not null)
+        {
+            tempFlights = tempFlights.Where(flight => flight.Class.Any(availableClass => availableClass.Value <= priceTo)).ToList();
+        }
+
+        return tempFlights;
+    }
+
+    public List<FlightModel.Flight> GetFlights()
+    {
+        return Flights;
+    }
+
+    public void DisplayFlights()
+    {
+        Console.WriteLine($"Available flights: ");
+
+        foreach (var flight in Flights)
+        {
+            Console.WriteLine(flight.ToString());
+        }
+    }
+
+    public void LoadFlights(string? path = null)
+    {
+        if (path is null) path = AppSettingsInitializer.AppSettingsInstance.FlightsRepoPath;
+        using (var reader = new StreamReader(path))
```

Comment:
Prefer a `using` declaration to avoid code nesting.

```csharp
using var reader = new StreamReader(path);
while (!reader.EndOfStream)
{
    var line = reader.ReadLine();
    var values = line.Split(',');
    ...
}
```